book.js

```diff
@@ -1,107 +1,295 @@
+
+//Creating step one modal
+export function displayModalStepOne(container, title) {
+  modalBackground.style.display = "block";
+  modalContainer1.style.display = "block";
+  modalTitle.setAttribute("id", "modal1__title");
+  modalSubTitle.textContent = "What date would you like to come?";
+  modalDate.textContent = "Date";
+  inputDate.value = currentDate;
+
+  modalContainer1.append(
+    modalTitle,
+    modalSubTitle,
+    modalDate,
+    inputDate,
+    errorDiv,
+    modalButtonSearch
+  );
+
+  modalBackground.append(modalContainer1);
+  const modal1 = modalBackground;
+  const modal1title = modalTitle;
+  return modal1, modal1title;
+}
+
+//Creating step two modal
+function displayModalStepTwo() {
+  modalTitle.textContent =
+    'Book Room: "' + selectedChallenge.data.title + '" (Step 2)';
+  modalBackground.style.display = "block";
+  modalContainer1.style.display = "none";
+  modalContainer2.style.display = "block";
+  modalTitle.setAttribute("id", "modal2__title");
+  modalName.textContent = "Name";
+  modalMail.textContent = "E-mail";
+  modalTime.textContent = "What time?";
+  modalParticipants.textContent = "How many participants?";
+
+  modalContainer2.append(
+    modalTitle,
+    errorDiv,
+    modalName,
+    inputName,
+    modalMail,
+    inputMail,
+    modalTime,
+    selecMenuTime,
+    modalParticipants,
+    selectMenuPart,
+    modalButtonSubmit
+  );
+  modalBackground.append(modalContainer2);
+}
+
+//eventlistener for list
+
+//Creating step three modal
+export async function displayModalStepThree() {
+  modalBackground.style.display = "block";
+  modalContainer2.style.display = "none";
+  modalContainer3.style.display = "flex";
+
+  modalTitleThankYou.textContent = "Thank you!";
+  modalLinkBack.text = "Back to challenges";
+  modalLinkBack.setAttribute("href", "challenges.html");
+
+  modalContainer3.append(modalTitleThankYou, modalLinkBack);
+  modalBackground.append(modalContainer3);
+}
+
+export async function availableTimes() {
+  modalButtonSearch.addEventListener("click", function () {
+    let params = new URLSearchParams(url.search);
+    let challengeID = selectedChallenge.data.id;
+    if (inputDate.value < currentDate || inputDate.value > withinNextYear) {
+      errorMsg.textContent =
+        "You must choose a date newer than today and within a year!";
+      errorMsg.style.color = "red";
+    } else {
+      params.set("date", inputDate.value);
+      params.set("id", parseInt(challengeID));
+
+      url.search = params.toString();
+      url = url.toString();
+
+      displayTimesAndParticipants();
+      displayModalStepTwo();
+      errorMsg.textContent = "";
+    }
+    return url;
   });
-});
+}
 
-//fetch challenge data
-const challengesArray = [];
-async function getChallengesData() {
-  const url = "https://lernia-sjj-assignments.vercel.app/api/challenges";
-  const res = await fetch(url);
+export async function displayTimesAndParticipants() {
+  let newUrl = url;
```

Comment:
I think I understand what you're trying to do (or at one point tried to do) here, but this line of code is actually useless. Javascript has two categories of values. Some values (often called "plain" or "simple") are always copied, while others are always "referenced".

A number is copied, so any new variable assigned the value of a different variable, will actually contain a _copy_ of that value. This is true for strings, booleans, and numbers:

```js
let x = 5;
let y = x;
y++; // This doesn't actually change the number 5, but replaces it with a 6
console.log(x, y); // Prints 5  6 (because x was unaffected by y++)
```

The other category are objects (including arrays, which are also objects) and they are always referenced. For example with an instance of the  `URL` class:

```js
let x = new URL('http://google.com');
let y = x; // Points to the same URL instance as x
y.search = '?foo=bar'; // Modifies both x and y
console.log(x, y) // Prints the same URL twice
```